### PR TITLE
UF-459: Preference API Refactor

### PR DESCRIPTION
**Pull request by adrielparedes**
Preference API changed, so I needed to make some minor changes here.

Dependes on: https://github.com/uberfire/uberfire/pull/605

### CODE REVIEW

**Comment by adrielparedes** [comment]
jenkins retest this

**Comment by csadilek** [comment]
jenkins retest this.

**Comment by paulovmr** [comment]
jenkins retest this

**Comment by adrielparedes** [comment]
jenkins retest this

File: kie-wb-common-stunner/kie-wb-common-stunner-showcase/kie-wb-common-stunner-showcase-project/pom.xml [hunk 1]
```diff
@@ -1018,6 +1018,7 @@
             <compileSourcesArtifact>org.uberfire:uberfire-client-api</compileSourcesArtifact>
             <compileSourcesArtifact>org.uberfire:uberfire-preferences-api</compileSourcesArtifact>
             <compileSourcesArtifact>org.uberfire:uberfire-preferences-client</compileSourcesArtifact>
+            <compileSourcesArtifact>org.uberfire:uberfire-preferences-ui-client</compileSourcesArtifact>
```
Comment (paulovmr) [inline kie-wb-common-stunner/kie-wb-common-stunner-showcase/kie-wb-common-stunner-showcase-project/pom.xml]: The dependency for `uberfire-preferences-ui-client` should be also added in this `pom.xml`. The other projects are already there.

Comment (adrielparedes) [inline kie-wb-common-stunner/kie-wb-common-stunner-showcase/kie-wb-common-stunner-showcase-project/pom.xml]: @paulovmr you are right, uploading now

Comment (adrielparedes) [inline kie-wb-common-stunner/kie-wb-common-stunner-showcase/kie-wb-common-stunner-showcase-project/pom.xml]: @paulovmr ready

**Comment by paulovmr** [comment]
jenkins retest this

**Pull request merged**